### ISSUE TITLE
Login callback - better way of handling errors

### ISSUE THREAD

**Issue by zepa19** [thecloudnativewebapp/GoCloudNative.Bff#21]
Hi, first of all, great job with this project.

I have one feature that could be added to the login callback endpoint.

I am using IdentityServer4. If the user consent was not given, then the user is redirected back to the login callback endpoint with `error=access_denied` query param.

Currently the BFF will display 400 page (BadRequest), because that callback only checks for presence of the `code` query param. It would be nice, to have a way to customize that behaviour. Idea - redirect to the specific SPA page (defined in configuration). The error query param should be provided in the redirection url to the SPA page as well

**Comment by appie2go** [comment]
Hi,

Thanks for your feedback. Happy to hear the project is useful to you!

About your issue: That's a good point. Basically, what you are asking is to add the ability to redirect to a specific, configurable endpoint after the authentication request has been completed?

I wil add that.

Cheers,

**Comment by appie2go** [comment]
Hi,

I have implented two new features:

___1. Configuring a custom authentication-error page___

Implement as follows:

```csharp
builder.Services.AddSecurityBff(o =>
    {
        o.ConfigureOpenIdConnect(oidcConfig, "oidc");
        o.SetAuthenticationErrorPage("/foo/oops");
        o.LoadYarpFromConfig(builder.Configuration.GetSection("ReverseProxy"));
    });
```

If you configure this, upon an error response from the IDP, the BFF redirects to the error path automatically. 

* You can only configure a relative url, you cannot redirect to another domain
* You cannot redirect to a page with a query-string or a # in the url
* Upon error, the BFF adds the querystring from the callback to the path you've configured (e.g. ?error=lorumipsum&errordescription=...&and=so&forth)
* When you do not configure anything, by default, the BFF will redirect to /account/login/callback/error

___2. Configuring a custom landing page___

It doesn't make too much sense either that it wasn't possible to configure a landing page either. So, I've implemented a feature which allows you to configure what page the BFF will redirect to after a succesful login:

```csharp
builder.Services.AddSecurityBff(o =>
    {
        o.ConfigureOpenIdConnect(oidcConfig, "oidc");
        o.SetLandingPage("/foo/welcome.html");
        o.LoadYarpFromConfig(builder.Configuration.GetSection("ReverseProxy"));
    });
```

* You can only configure a relative url, you cannot redirect to another domain
* By default the BFF redirects to /

Let me know if implementing these two features as described solves your problem. Then I'll make a release.

Cheers,

**Comment by zepa19** [comment]
Great, thats exactly what I needed.

**Comment by thecloudnativewebapp** [comment]
Great. I've just released v0.0.13. When you update to this version, you can use the functionality.

Closing the issue now.

Cheers!
